tests/src/test_projects/storagemapvec/src/main.sw

```diff
@@ -0,0 +1,68 @@
+contract;
+
+use sway_libs::storagemapvec::StorageMapVec;
+
+storage {
+    mapvec: StorageMapVec<u64, u64> = StorageMapVec {},
+}
+
+abi TestContract {
+    #[storage(read, write)]
+    fn push(key: u64, value: u64);
+
+    #[storage(read, write)]
+    fn pop(key: u64) -> Option<u64>;
+
+    #[storage(read)]
+    fn get(key: u64, index: u64) -> Option<u64>;
+
+    #[storage(read)]
+    fn len(key: u64) -> u64;
+
+    #[storage(read)]
+    fn is_empty(key: u64) -> bool;
+
+    #[storage(write)]
+    fn clear(key: u64);
+
+    #[storage(read)]
+    fn to_vec_as_tup(key: u64) -> (u64, u64, u64);
+}
+
+impl TestContract for Contract {
+    #[storage(read, write)]
+    fn push(key: u64, value: u64) {
+        storage.mapvec.push(key, value);
```

Comment:
can you add a swap?

Reply:
Can you give me an example of what you want exactly? Do you want to swap entire vectors? Or items inside a vector

Reply:
the use case is a `remove(index)` in O(1)
accomplished by `swap(index, len-1)` and `pop()`

Reply:
I have added a swap_remove aswell as swap, should be more efficient